### DISCUSSION TITLE
Composio with AutoGen 0.4

### DISCUSSION THREAD

**Discussion by ChampPhil** [microsoft/autogen#4330] [Q&A]
Hello!

Is there a way to support autogen0.4 with composio?
The 'composio-autogen' package supports default autogen (v0.2) but runs into an error with 0.4.

Here is the error message from the console, along with a screenshot of the .py file.
(This is in a Github Codespace)
      
<img width="564" alt="code" src="https://github.com/user-attachments/assets/fb304a92-a39d-4afd-b693-eb007e4ec70d">
<img width="652" alt="error" src="https://github.com/user-attachments/assets/beea7fd8-10dd-47b1-a63e-06b400f1848f">

**Comment by ekzhu** [comment]
We would love to have Composio support AutoGen v0.4. @sawradip

This should become a community packages: https://microsoft.github.io/autogen/dev/user-guide/extensions-user-guide/index.html

**Reply by ChampPhil** [reply]
I completely agree. I feel like, because multi-agent systems (like Autogen) and these comprehensive toolkits (like Composio) have really amazing synergy, AutoGen v0.4. supporting Composio would be amazing!

**Comment by LuDraGa** [comment]
Has anyone used the provider setup for composio to build this ??
https://docs.composio.dev/providers/custom/python
